hr_timesheet_retroactive_cost/models/hr_employee_timesheet_cost.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class HrEmployeeTimesheetCost(models.Model):
+    _name = "hr.employee.timesheet.cost"
+    _description = "Employee timesheet cost history"
+    _order = "create_date DESC"
+
+    employee_id = fields.Many2one(comodel_name="hr.employee")
+    currency_id = fields.Many2one("res.currency")
+    timesheet_cost = fields.Monetary(currency_field="currency_id")
+    retroactive_cost_date = fields.Date(
+        help="Change de timesheet cost from this date to now.",
+        default=fields.Date.context_today,
+    )
+
+    def _update_employee_history_cost(self, employee, currency, cost, date):
+        """Saves the cost change and deletes deprecated cost if the new record
+        overlaps old dates."""
+        cost_ids = self.search([("employee_id", "=", employee.id)])
+        cost_to_delete_ids = cost_ids.filtered(
+            lambda cost_id: cost_id.retroactive_cost_date >= date
+        )
```

Comment:
**issue(performance):** just add one more condition to the domain, and you don't need to re-filter later.